### ISSUE TITLE
Add `accept: application/gzip,application/octet-stream` header when downloading a chart

### ISSUE THREAD

**Issue by lwj5** [helm/helm#9507]
Hi,

It seems that the current HTTP/HTTPS request made to download a chart from an URL does not specify an `accept` header. This is an issue as some REST APIs serve multiple content types. Since we are specifically requesting a tar.gz (binary) file, the content type should be `application/gzip,application/octet-stream`.

For example, the popular Github API when downloading release binary requires such an `Accept` header, see documentation: https://docs.github.com/en/rest/reference/repos#get-a-release-asset.

This is not limited to Github API but also any other APIs that serve multiple content types.

TLDR add header `accept: application/gzip,application/octet-stream` to HTTPClient when getting charts.

**Comment by jdolitsky** [comment]
Hi thanks. I'm unsure if doing this for all repos would cause people to have issues.

**Comment by lwj5** [comment]
Hmmm, one way to go about it is to use the header `accept: application/gzip,application/octet-stream` and fallback to no headers with a warning message if the request returns an error.

According to https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types, I do not see any other mime types that helm should be accepting, since helm is specifically looking for a gzip file.

But it's up for discussion
